thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -360,6 +353,7 @@ class CDistributorBase : public CSimpleInterface, implements IHashDistributor, i
             {
                 return bucket.getClear();
             }
+            atomic_t *querySenderFinished() { return &senderFinished; }
```

Comment:
minor: To improve encapsulation, slightly better would be to export a queryMarkSenderFinished() function which performs the cas and returns the boolean.  Then the knowledge of the atomic_t stays inside this class.